client/navigation/components/Item/index.js

```diff
@@ -1,47 +1 @@
-/**
- * External dependencies
- */
-import { NavigationItem, useSlot } from '@woocommerce/experimental';
-import { recordEvent } from '@woocommerce/tracks';
-import { WooNavigationItem } from '@woocommerce/navigation';
-
-const Item = ( { item } ) => {
-	const slot = useSlot( 'woocommerce_navigation_' + item.id );
-	const hasFills = Boolean( slot?.fills?.length );
-
-	const trackClick = ( id ) => {
-		recordEvent( 'navigation_click', {
-			menu_item: id,
-		} );
-	};
-
-	// Disable reason: The div wrapping the slot item is used for tracking purposes
-	// and should not be a tabbable element.
-	/* eslint-disable jsx-a11y/no-static-element-interactions, jsx-a11y/click-events-have-key-events */
-
-	// Only render a slot if a coresponding Fill exists and the item is not a category
-	if ( hasFills && ! item.isCategory ) {
-		return (
-			<NavigationItem key={ item.id } item={ item.id }>
-				<div onClick={ () => trackClick( item.id ) }>
-					<WooNavigationItem.Slot name={ item.id } />
-				</div>
-			</NavigationItem>
-		);
-	}
-
-	return (
-		<NavigationItem
-			key={ item.id }
-			item={ item.id }
-			title={ item.title }
-			href={ item.url }
-			navigateToMenu={ ! item.url && item.id }
-			onClick={ () => trackClick( item.id ) }
-			hideIfTargetMenuEmpty
-		/>
-	);
-	/* eslint-enable jsx-a11y/no-static-element-interactions, jsx-a11y/click-events-have-key-events */
-};
-
-export default Item;
+export * from './item';
```

Comment:
Is there a way to preserve the Git history on this better? 

I think it would need the the original `index.js` file moved to `item.js` first, then creating a new `index.js`. I'm not sure if you did this or if GitHub just isn't showing the diff accurately?

Reply:
Thanks Adrian. This was my intention—I think I'll have to experiment with how Github handles these diffs to understand if I did this correctly 👍🏻

Reply:
Just a follow-up on this. After a little limited testing I don't believe there is a way to execute this while retaining a more straight-forward history. I've tried using `git mv`, moving manually, and even separating file creation and file renaming into separate commits. Same results.

There are a few threads scattered around alluding to this, such as [this one](https://stackoverflow.com/questions/2314652/is-it-possible-to-move-rename-files-in-git-and-maintain-their-history). I _have_ had luck instructing git to follow renames via various tools however, as mentioned in the [related P2 post](https://teamisotope.wordpress.com/2021/08/12/a-proposal-for-moving-forward-with-naming-conventions-in-wc-admin/).